doc/mainpage.dox

```diff
@@ -223,5 +223,106 @@ before the measurement framework or in a particular Plugin/Algorithm not knowing
 failure modes.  This means that Plugin/Algorithm implementations should generally have a global try/catch
 block that re-throwns lower-level exceptions corresponding to known failure modes as @ref MeasurementErrors.
 
+@section measBaseFlagHandlerInPython Using a FlagHandler with Python Plugins
+
+As mentioned above, a plugin measure() method is used to implement the code which performs
+measurements on individual detected sources.  It also records the results in the output
+source catalog. If an error occurs during measure(), the plugin is obliged to record the error by setting a
+flag in the source catalog.  As described above, this is often done by raising a MeasurementError and relying
+on the fail() method of the plugin to set the error flags.
+
+<B>
+How to Define a FlagHandler in Python:
+</B>
+
+If you have looked at any of the C++ implementations of plugins in meas_base, you will notice that the C++
+algorithms implement their fail() method using a C++ class called lsst::meas::base::FlagHander.  In this
+section, we will show you how to do the same thing in Python using the addFlagHandler decorator.
+
+First examine the code testFlagHandler.py in the meas_base tests directory. This unit test defines a
+plugin written in Python which illustrates several useful techniques about using the FlagHander.
+Note first the line decorating the class definition, which reads like this:
+
+@code{.py}
+@addFlagHandler(("flag", "General Failure error"),
+                ("flag_containsNan","Measurement area contains a nan"),
+                ("flag_edge", "Measurement area over edge"))
+@endcode
+This decorator modifies your plugin class to add a member variable named "flagHandler". As you might expect,
+this is an object of type lsst.meas.base.FlagHandler which you can use to manage errors. The FlagHandler
+will define the following:
+
+- a general failure flag which indicates that something has gone wrong during measure().
+- a specific failure flag which indicates a source which contain nans.
+- a specific failure flag which indicates that the source is too close to the edge to be measured.
+
+The FlagHander makes it simple to implement your fail() method in Python. Recall that you must implement
+this method in you plugin class if your method can fail: either with an expected error which your code 
+raises as a MeasurementError, or an unexpected error which is caught by the measurement framework.
+
+The following addition to your class will correctly implement the error handling:
+
+@code{.py}
+    def fail(self, measRecord, error=None):
+        if error is None:
+            self.flagHandler.handleFailure(measRecord)
+        else:
+            self.flagHandler.handleFailure(measRecord, error.cpp)
+@endcode
+
+<B>
+Using the ErrEnum in Error Handling:
+</B>
+
+In addition to the flagHander member, the decorator also adds an "ErrEnum" member to your plugin.
+This enumeration allows each of the errors to be simply and uniquely identified.
```

Comment:
I'd add a sentence saying that the names of errors available from errEnum are those added via the addFlagHandler class method.